security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/PersistableConfiguration.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.oauth2.endpoint;
+
+import io.micronaut.core.util.Toggleable;
+
+import java.util.Optional;
+
+/**
+ * Base configuration for persistable endpoints.
+ *
+ * @author Álvaro Sánchez-Mariscal
+ */
+public interface PersistableConfiguration extends Toggleable {
+
+    /**
+     * @return The nonce persistence mechanism
+     */
+    Optional<String> getPersistence();
```

Comment:
annotate return type with `@NonNull`

Reply:
The interfaces that were refactored into `PersistableConfiguration` (`NonceConfiguration` and `StateValidationConfiguration`) weren't annotated with `@NonNull` either. But ok, will add it.